src/xcode/ENA/ENA/Source/Services/Exposure Transaction/__tests__/ExposureDetectionTests.swift

```diff
@@ -75,18 +89,67 @@ final class ExposureDetectionTransactionTests: XCTestCase {
 
 		wait(
 			for: [
+				supportedCountriesToBeCalled,
 				availableDataToBeCalled,
 				downloadDeltaToBeCalled,
 				downloadAndStoreToBeCalled,
-				configurationToBeCalled,
 				writtenPackagesBeCalled,
+				configurationToBeCalled,
 				summaryResultBeCalled,
 				startCompletionCalled
 			],
 			timeout: 1.0,
 			enforceOrder: true
 		)
 	}
+
+	func test_When_NoRemoteDataAvailable_Then_FailureNoDaysAndHoursIsCalled() {
+		let delegate = ExposureDetectionDelegateMock()
+
+		delegate.availableData = {
+			return nil
+		}
+
+		let packageDownloader = CountryKeypackageDownloader(delegate: delegate)
+
+		let detection = ExposureDetection(
+			delegate: delegate,
+			countryKeypackageDownloader: packageDownloader
+		)
+
+		let expectationNoDaysAndHours = expectation(description: "completion with NoDaysAndHours error called.")
+
+		packageDownloader.downloadKeypackages(for: "DE") { result in
+			switch result {
+			case .failure(let error):
+				switch error {
+				case .noDaysAndHours:
+					expectationNoDaysAndHours.fulfill()
+				default:
+					XCTFail("noDaysAndHours error expteced.")
+				}
+			case .success:
+				XCTFail("downloadKeypackages should failt due to missing data.")
+			}
+		}
+
+		let expectationDetectionCompletion = expectation(description: "Detection completion was called.")
+		detection.start { _ in
+			expectationDetectionCompletion.fulfill()
+		}
+
+		waitForExpectations(timeout: 1.0)
+	}
+
+	func makeCountries() -> [Country] {
+		guard let enCountry = Country(countryCode: "FR"),
```

Comment:
nitpick: frCountry? 
could also be shortened to something like:
`[Country(countryCode: "FR"), Country(countryCode: "IT")].compactMap { $0 }`
(which does not XCFail() when the country init fails but this should be covered by other tests..)

Reply:
changed